client/dashboard/sites/performance/index.tsx

```diff
@@ -148,7 +158,29 @@ function SitePerformance() {
 			overlay={ <PageLayout header={ <PageHeader /> } /> }
 			{ ...getPerformanceCalloutProps() }
 		>
-			<SitePerformanceContent site={ site } />
+			{ site.is_coming_soon || site.is_private ? (
+				<PageLayout
+					size="small"
+					header={ <PageHeader /> }
+					notices={
+						<Notice
+							title={ __( 'Launch your site to start measuring performance' ) }
+							actions={
+								<SiteLaunchButton
+									site={ site }
+									onClick={ () =>
+										recordTracksEvent( 'calypso_dashboard_performance_launch_site_click' )
```

Comment:
Maybe it should be a `tracksName` prop that accepts the event name to record? That way we can make it a non-optional requirement. You _have_ to remember to have an event for it.

Reply:
Introduced the `tracksContext` by https://github.com/Automattic/wp-calypso/pull/106293/commits/bca133194c992508a30c33ddbcb29b75bf9b0f95 to track where the event is triggered